modules/otel-troubleshoot-instrumentation.adoc

```diff
@@ -0,0 +1,91 @@
+
+If the Instrumentation resource is not available when the application is deployed, the operator will not inject the auto-instrumentation. When you want to update the instrumentation settings, you need to delete the instrumentation object and redeploy the application.
+
+To verify that the `opentelemetry-auto-instrumentation`` init-container has successfully started, execute the following command:
+
+[source,console]
+----
+oc get events -n <your-project>
+----
+
+You should see something like this:
+
+[source,console]
+----
+... Created container opentelemetry-auto-instrumentation
+... Started container opentelemetry-auto-instrumentation
+----
+
+If those messages are missing, then it means that was an issue while injecting the auto-instrumentation.
+
+. Inspect for the operator logs for instrumentation errors:
++
+[source,console]
+----
+oc logs -l app.kubernetes.io/name=opentelemetry-operator --container manager -n openshift-opentelemetry-operator --follow
+----
+
+. Confirm that the application pods you want to instrument are labeled correctly and the appropriate auto-instrumentation settings have been applied. For example, if you are instrumenting a Python application, you can check that the pods have the `instrumentation.opentelemetry.io/inject-python='true'` annotation:
++
+[source,console]
+----
+oc get pods -n <your-project> -o jsonpath='{range .items[?(@.metadata.annotations["instrumentation.opentelemetry.io/inject-python"]=="true")]}{.metadata.name}{"\n"}{end}'
+----
+
+You should double-check the annotation applied to the instrumentation object is the correct one for the language you are instrumenting. If there are multiple instrumentations in the same namespace, you should set its name in the annotation. Also, if the Instrumentation object is in a different namespace, you need to specify it in the annotation. You can find the accepted values in the link:otel-instrumentation.html["OpenTelemetry instrumentation configuration options"] section.
+
+When defining a Deployment, remember that annotations can be added in either {{spec.metadata.annotations}} or {{spec.template.metadata.annotations}}. Ensure that the auto-instrumentation annotation is added to {{spec.template.metadata.annotations}}.
+
+
+== Check the instrumentation libraries are generating telemetry data
+. Verify the endpoint that the instrumentation is transmitting data to:
++
+[source,console]
+----
+oc get instrumentation <instrumentation-name> -n <your-project> -o jsonpath='{.spec.endpoint}'
+----
++
+By default, the endpoint is `http://localhost:4317`. This will work if you deployed the OpenTelemetry Collector as a sidecar container in your application pod. If you are using a different endpoint, you can update it by editing the instrumentation object and redeploying your application.
```

Comment:
Regarding the sentence "This will work if you deployed the OpenTelemetry Collector as a sidecar container in your application pod."

- Is it that "This will [also] work if you deployed the OpenTelemetry Collector as a sidecar container in your application pod."
- Or is it that "This will [only] work if you deployed the OpenTelemetry Collector as a sidecar container in your application pod."
?

Reply:
This will only work [...]

Reply:
This [command?] will only work ...?

Reply:
It is not the command. It is:
```
By default, the endpoint is `http://localhost:4317`. Sending telemetry data will only work with the default value properly if you deployed the OpenTelemetry Collector as a sidecar container in your application pod.
```

Reply:
Fixed. Thank you.